app/scripts/modules/core/src/orchestratedItem/orchestratedItem.transformer.ts

```diff
@@ -107,6 +108,44 @@ export class OrchestratedItemTransformer {
     return null;
   }
 
+  private static getLockFailureException(task: ITask): string {
+    const generalException: any = task.getValueFor('exception');
+    if (generalException) {
+      const details: any = generalException.details;
+      if (details && details.currentLockValueApplication && details.currentLockValueType && details.currentLockValue) {
+        let typeDisplay: string;
+        let linkUrl: string;
+
+        if (details.currentLockValueType === 'orchestration') {
+          typeDisplay = 'task';
+          linkUrl = ReactInjector.$state.href('home.applications.application.tasks.taskDetails', {
+            application: details.currentLockValueApplication,
+            taskId: details.currentLockValue
+          });
+        } else {
+          typeDisplay = 'pipeline';
+          linkUrl = ReactInjector.$state.href('home.applications.application.pipelines.executionDetails.execution', {
+            application: details.currentLockValueApplication,
+            executionId: details.currentLockValue
+          });
+        }
+
+        return 'Failed to acquire lock. An <a href="' + linkUrl + '">existing ' + typeDisplay + '</a> is currently operating on the cluster.';
+      }
+    }
+    return null;
+  }
+
+  private static getCustomException(task: ITask): string {
+    const generalException: any = task.getValueFor('exception');
+    if (generalException) {
+      if (generalException.exceptionType && generalException.exceptionType === 'LockFailureException') {
```

Comment:
a hook point to key error message rendering on specific exception types (with one specific type for LockFailureException)